src/Repositories/FilesystemRepository.php

```diff
@@ -43,6 +43,31 @@ public function get(string $path): ?Sheet
         });
     }
 
+    public function getBySlug(string $slug): ?Sheet
+    {
+        $parts = explode('/', $slug);
+
+        $name = array_pop($parts);
+
+        $pattern = sprintf(
+            '/^%s%s%s$/',
+            preg_quote(implode('/', $parts), '/'),
+            !empty($parts) ? preg_quote('/', '/') : '',
+            '([\w\-]+\.|)'.preg_quote($this->normalizePath($name))
+        );
+
+        return $this->get(
+            collect($this->filesystem->allFiles())
+                ->filter(function (string $path): bool {
+                    return Str::endsWith($path, ".{$this->extension}");
+                })
+                ->filter(function (string $path) use ($pattern): bool {
```

Comment:
Instead of `->filter(callback)->filter(callback)->first()`, maybe `->first(callback)` would be better here. Slightly less readable because the callback needs two clauses ([1]an early return for the extension, [2] the regex matching), but it could have a significant performance impact. Especially if we'd combine it with lazy collections in the near future.